api/api.go

```diff
@@ -68,12 +112,30 @@ func (binding *SplitioAPIBinding) httpGet(path string, segment string, since int
 	return data, nil
 }
 
-// GetSegmentChanges will get segment data
-func (binding *SplitioAPIBinding) GetSegmentChanges() error {
-	return fmt.Errorf("not implemented")
-}
+// getAllChanges polls the Split.io API until since and till are the same
+// path is the path of the HTTP request e.g "splitChanges", "segmentChanges"
+// segment is the segment name, will be empty when the end point is splitChanges
+func (binding *SplitioAPIBinding) getAllChanges(path string, segment string) ([]map[string]interface{}, int64, error) {
```

Comment:
`segment` doesn't mean anything when fetching splits.  What if you update this function to take only a `path` instead?

Reply:
Yeah I was thinking to do that. But after trying the `"/"` will be encoded to `"%2F"` so I got the 404 error. (https://github.com/godaddy/split-go-serializer/blob/master/api/api.go#L36)
e.g the url will become: 
```
https://sdk.split.io/api/getSegments%2Ftest-segment/?since=-1
```
not
```
https://sdk.split.io/api/getSegments/test-segment/?since=-1
```

Reply:
don't encode those parameters then.  e.g.,

```go
Get(fmt.Sprintf("%s/%s", binding.splitioAPIUri, path))
```